## Шаг 1. Получение данных

In [1]:
# импорт библиотек
import pandas as pd
from pymystem3 import Mystem
from collections import Counter 
import warnings 
warnings.filterwarnings('ignore')

In [2]:
# чтение файла с данными и сохранение в clients 
try:
    clients = pd.read_csv('/datasets/data.csv')
except FileNotFoundError:
    print('файл или директория не существует!')

In [3]:
# функция получения информации
def get_info(name, df): 
    print('\033[1m' + 'Первые 5 строк таблицы: ' + name + '\033[0m')
    display(df.head())
    print()
    print('\033[1m' + 'Сводная информация о таблице: ' + name + '\033[0m')
    print()
    df.info()
    print()
    print('\033[1m' + 'Описательная статистика таблицы: ' + name + '\033[0m')
    display(df.describe())
    print()
    print('\033[1m' + 'Описательная статистика таблицы: ' + name + '\033[0m')
    display(df.describe(include=[object]))
    print()
    print('\033[1m' + 'Пропуски в таблице: ' + name + '\033[0m')
    report = df.isna().sum().to_frame()
    report = report.rename(columns = {0: 'missing_values'})
    report['% of total'] = (100 * report['missing_values'] / df.shape[0]).round(2)
    display(report.sort_values(by = 'missing_values', ascending = False))
    print()
    print('\033[1m' + 'Дубликатов в таблице ' + name + '\033[0m'':', df.duplicated().sum())
    print()

In [4]:
# получаем информацию по таблицам
dict_df = {'Клиенты': clients} 
for dfd in dict_df:
    get_info(dfd, dict_df[dfd])

Первые 5 строк таблицы: Клиенты


children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу


Сводная информация о таблице: Клиенты

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB

Описательная статистика таблицы: Клиенты


children  days_employed     dob_years  education_id  \
count  21525.000000   19351.000000  21525.000000  21525.000000   
mean       0.538908   63046.497661     43.293380      0.817236   
std        1.381587  140827.311974     12.574584      0.548138   
min       -1.000000  -18388.949901      0.000000      0.000000   
25%        0.000000   -2747.423625     33.000000      1.000000   
50%        0.000000   -1203.369529     42.000000      1.000000   
75%        1.000000    -291.095954     53.000000      1.000000   
max       20.000000  401755.400475     75.000000      4.000000   

       family_status_id          debt  total_income  
count      21525.000000  21525.000000  1.935100e+04  
mean           0.972544      0.080883  1.674223e+05  
std            1.420324      0.272661  1.029716e+05  
min            0.000000      0.000000  2.066726e+04  
25%            0.000000      0.000000  1.030532e+05  
50%            0.000000      0.000000  1.450179e+05  
75%            1.000000      0.000000  2.034351e+05  
max            4.000000      1.000000  2.265604e+06


Описательная статистика таблицы: Клиенты


education    family_status gender income_type  purpose
count      21525            21525  21525       21525    21525
unique        15                5      3           8       38
top      среднее  женат / замужем      F   сотрудник  свадьба
freq       13750            12380  14236       11119      797


Пропуски в таблице: Клиенты


missing_values  % of total
days_employed               2174        10.1
total_income                2174        10.1
children                       0         0.0
dob_years                      0         0.0
education                      0         0.0
education_id                   0         0.0
family_status                  0         0.0
family_status_id               0         0.0
gender                         0         0.0
income_type                    0         0.0
debt                           0         0.0
purpose                        0         0.0


Дубликатов в таблице Клиенты: 54



В столбце 'children' отрицательные значения и аномально большое значение 20.

In [5]:
clients[clients['children'] < 20]['children'].describe() # получение описательной статистики столбца 'children' без максимального значения

count    21449.000000
mean         0.469952
std          0.754165
min         -1.000000
25%          0.000000
50%          0.000000
75%          1.000000
max          5.000000
Name: children, dtype: float64

Так как нет значений между 5 и 20, скорее всего 20 это опечатка.

В столбце 'days_employed' есть отрицательные значения и аномально большие значения.

В столбце 'dob_years' есть значения 0. 

In [6]:
clients['education'].unique() # просмотр уникальных значений в столбце education

array(['высшее', 'среднее', 'Среднее', 'СРЕДНЕЕ', 'ВЫСШЕЕ',
       'неоконченное высшее', 'начальное', 'Высшее',
       'НЕОКОНЧЕННОЕ ВЫСШЕЕ', 'Неоконченное высшее', 'НАЧАЛЬНОЕ',
       'Начальное', 'Ученая степень', 'УЧЕНАЯ СТЕПЕНЬ', 'ученая степень'],
      dtype=object)

В столбце 'education' разный регистр. 

In [7]:
clients['family_status'].unique() # просмотр уникальных значений в столбце family_status

array(['женат / замужем', 'гражданский брак', 'вдовец / вдова',
       'в разводе', 'Не женат / не замужем'], dtype=object)

In [8]:
clients['gender'].unique() # просмотр уникальных значений в столбце gender

array(['F', 'M', 'XNA'], dtype=object)

В столбце 'gender' неизвестное значение XNA.

In [9]:
clients['purpose'].unique() # Просмотр уникальных значений в столбце purpose

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Вывод**

При изучении файла выявлены ошибки:
1. В столбце 'children' отрицательные значения и аномально большое значение 20. Отрицательного количества детей быть не может, вернём абсолютные значения. 20 получилось из-за одновременного нажатия клавиш '2' и '0' заменим на 2.
2. В столбце 'days_employed' есть пропуски и отрицательные значения, значения больше 0 аномально большие, скорее всего это в часах, поделим на 24, меньше 0 вернём абсолютные значения. Заменим тип данных на целочисленный.
3. В столбце 'dob_years' есть 0 значения.
4. В столбце 'education' строки в разных регистрах, приведём к нижнему регистру.
5. В столбце 'gender' неизвестное значение XNA, заменим на самый часто встречающейся.
6. В столбце 'total_income' есть пропуски.
7. В столбце 'purpose' есть много разных формулировок, которые можно категоризировать.
8. Возможны две причины возникновения пропусков:
- Технический сбой. (случайные пропуски)
- Клиенты не предоставили документы подтверждающие доход и стаж. (не случайные)

Перед анализом нужно предобработать данные.





## Шаг 2. Предобработка данных

### Обработка пропусков

В столбце days_employed 2174 пропуска, перед заполнением преобразуем данные.

In [10]:
clients['days_employed'] = clients['days_employed'].apply(abs) # возвращаем абсолютные значения  
clients.loc[clients['days_employed'] >27375, 'days_employed'] = clients.loc[clients['days_employed'] >27375, 'days_employed'] / 24 # делим на 24 все занчения больше 27375 (75*365)

In [11]:
for income_type in clients.loc[clients['days_employed'].isna()]['income_type'].unique(): # создадим цикл перебирающий уникальные значения типов занятости в строках где есть пропуски 
    clients.loc[(clients['days_employed'].isna()) & (clients['income_type'] == income_type),'days_employed'] = clients[clients['income_type'] == income_type]['days_employed'].median() # заполняем пропуски медианным значением в зависимости от типа занятости
    clients.loc[(clients['total_income'].isna()) & (clients['income_type'] == income_type),'total_income'] = clients[clients['income_type'] == income_type]['total_income'].median() # заполняем пропуски медианным значением в зависимости от типа занятости
    print(income_type, 'стаж:', clients[clients['income_type'] == income_type]['days_employed'].median(), 
          'зарплата:', clients[clients['income_type'] == income_type]['total_income'].median())

пенсионер стаж: 15217.221094405468 зарплата: 118514.48641164352
госслужащий стаж: 2689.3683533043886 зарплата: 150447.9352830068
компаньон стаж: 1547.3822226779334 зарплата: 172357.95096577113
сотрудник стаж: 1574.2028211070854 зарплата: 142594.39684740017
предприниматель стаж: 520.8480834953765 зарплата: 499163.1449470857


Убедимся, что в таблице не осталось пропусков. Для этого ещё раз посчитаем пропущенные значения.

In [12]:
clients.isna().sum() # подсчёт пропусков

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

**Вывод**

* В файле с данными были пропуски в одних и тех же строках столбцов days_employed и total_income.  
* Возможны две причины возникновения пропусков:
1. Технический сбой.
2. Клиенты не предоставили документы подтверждающие доход и стаж.
* Все пропуски были заполнены медианными значениями соответсвующего столбца.


### Замена типа данных

In [13]:
clients['children'] = clients['children'].apply(abs) # возвращаем абсолютные значения 
clients.loc[clients['children'] == 20, 'children'] = 2 # все значения в столбце children равные 20 заменяем на 2

In [14]:
clients['days_employed'] = clients['days_employed'].astype('int') # заменим вещественный тип данных на целочисленный

In [15]:
clients['total_income'] = (clients['total_income']/1000).astype('int') # заменим вещественный тип данных на целочисленный

In [16]:
clients['education'] = clients['education'].str.lower() # переводим все строки столбца education в нижний регистр
clients['education'].unique() # просмотр уникальных значений в столбце education

array(['высшее', 'среднее', 'неоконченное высшее', 'начальное',
       'ученая степень'], dtype=object)

In [17]:
clients.loc[clients['gender'] == 'XNA', 'gender'] = clients['gender'].sort_values()[0] # заменим строки содержащие значения XNA в столбце gender на самый часто встречающеся пол среди клиентов  
clients['gender'].unique()  # просмотр уникальных значений в столбце gender

array(['F', 'M'], dtype=object)

В столбце dob_years есть значения 0, т.к. новорожденнуму кредит не дадут скорее всего это технический сбой. Заменим 0 на значения зависящие от уровня образования и стажа.

In [18]:
clients[clients['dob_years'] == 0]['education'].unique() # просмотр уникальных значений в столбце education в тех строках где в столюце dob_years значение 0 

array(['среднее', 'высшее', 'неоконченное высшее'], dtype=object)

Значения 0 в столбцен dob_years всего у терёх уровней образования, к стажу в годах будем прибавлять для среднего 18, для неоконченного высшего 20, для высшего 20.


In [19]:
clients.loc[(clients['dob_years'] == 0) & (clients['education'] == 'среднее'), 'dob_years'] = clients['days_employed']/365+18
clients.loc[(clients['dob_years'] == 0) & (clients['education'] == 'неоконченное высшее'), 'dob_years'] = clients['days_employed']/365+20
clients.loc[(clients['dob_years'] == 0) & (clients['education'] == 'высшее'), 'dob_years'] = clients['days_employed']/365+22
clients['dob_years'] = clients['dob_years'].astype('int') # заменим вещественный тип данных на целочисленный

In [20]:
clients.describe() # получение описательной статистики

children  days_employed    dob_years  education_id  \
count  21525.000000   21525.000000  21525.00000  21525.000000   
mean       0.479721    4600.060720     43.44288      0.817236   
std        0.755528    5354.354946     12.28752      0.548138   
min        0.000000      24.000000     18.00000      0.000000   
25%        0.000000    1025.000000     33.00000      1.000000   
50%        0.000000    1993.000000     43.00000      1.000000   
75%        1.000000    5347.000000     53.00000      1.000000   
max        5.000000   18388.000000     75.00000      4.000000   

       family_status_id          debt  total_income  
count      21525.000000  21525.000000  21525.000000  
mean           0.972544      0.080883    164.721301  
std            1.420324      0.272661     98.044328  
min            0.000000      0.000000     20.000000  
25%            0.000000      0.000000    107.000000  
50%            0.000000      0.000000    142.000000  
75%            1.000000      0.000000    195.000000  
max            4.000000      1.000000   2265.000000

**Вывод**

* В столбцах days_employed, dob_years и total_income заменен вещественный тип данных на целочисленный методом astype.

### Обработка дубликатов

In [21]:
clients[clients.duplicated(keep = False)].sort_values(by = list(clients.columns))['days_employed'].unique() # просмотр уникальных значений в столбце days_employed у дублткатов 

array([ 1547,  1574, 15217,  2689])

In [22]:
clients = clients.drop_duplicates() # удаление дубликатов

In [23]:
clients.duplicated().sum() # подсчёт дубликатов

0

**Вывод**

* Найдено 71 дубликатов методом duplicate.
* Все дубликаты были удалены методом drop_duplicates.
* Во всех дубликатах были пропущены значения в столбцах days_employed и total_income, скорее всего это технический сбой.   

### Лемматизация

Для отвена на вопрос: как разные цели кредита влияют на его возврат в срок? нужно лемматизировать цели.

In [24]:
m = Mystem() # создаем объект-лемматизатор
purpose_uniq = clients['purpose'].unique() # создание списка уникальных целей кредита
purpose_uniq_str =' '.join(purpose_uniq) # преобразование списка в строку
lemmas = m.lemmatize(purpose_uniq_str) # лемматизируем строку
print(Counter(lemmas)) # подсчёт и вывод на экран слов встречающихся в строке

Counter({' ': 96, 'покупка': 10, 'недвижимость': 10, 'автомобиль': 9, 'образование': 9, 'жилье': 7, 'с': 5, 'операция': 4, 'на': 4, 'свой': 4, 'свадьба': 3, 'строительство': 3, 'получение': 3, 'высокий': 3, 'дополнительный': 2, 'для': 2, 'коммерческий': 2, 'жилой': 2, 'подержать': 2, 'заниматься': 2, 'сделка': 2, 'приобретение': 1, 'сыграть': 1, 'проведение': 1, 'семья': 1, 'собственный': 1, 'со': 1, 'профильный': 1, 'сдача': 1, 'ремонт': 1, '\n': 1})


Чаще всего в цклях встречаются существительные: недвижимость, автомобиль, образование, жилье и свадьба.

In [25]:
def purpose_lemm (row_values): # создаём функцию
    row_lemmas = m.lemmatize(row_values) # лемматизируем row_values
    if 'жилье' in row_lemmas:
        return 'недвижимость'
    if 'недвижимость' in row_lemmas:
        return 'недвижимость'
    if 'автомобиль' in row_lemmas:
        return 'автомобиль'
    if 'образование' in row_lemmas:
        return 'образование'
    if 'свадьба' in row_lemmas:
        return 'свадьба'
    return 'прочее'

clients['purpose_lemmas'] = clients['purpose'].apply(purpose_lemm) # создаём столбец purpose_lemmas и присваеваем категории 

/tmp/ipykernel_173/3281638371.py:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  clients['purpose_lemmas'] = clients['purpose'].apply(purpose_lemm) # создаём столбец purpose_lemmas и присваеваем категории


In [26]:
display(clients.head(10)) # получение первых 10 строк таблицы clients

children  days_employed  dob_years education  education_id  \
0         1           8437         42    высшее             0   
1         1           4024         36   среднее             1   
2         0           5623         33   среднее             1   
3         3           4124         32   среднее             1   
4         0          14177         53   среднее             1   
5         0            926         27    высшее             0   
6         0           2879         43    высшее             0   
7         0            152         50   среднее             1   
8         2           6929         35    высшее             0   
9         0           2188         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0           253   
1   женат / замужем                 0      F   сотрудник     0           112   
2   женат / замужем                 0      M   сотрудник     0           145   
3   женат / замужем                 0      M   сотрудник     0           267   
4  гражданский брак                 1      F   пенсионер     0           158   
5  гражданский брак                 1      M   компаньон     0           255   
6   женат / замужем                 0      F   компаньон     0           240   
7   женат / замужем                 0      M   сотрудник     0           135   
8  гражданский брак                 1      F   сотрудник     0            95   
9   женат / замужем                 0      M   сотрудник     0           144   

                      purpose purpose_lemmas  
0               покупка жилья   недвижимость  
1     приобретение автомобиля     автомобиль  
2               покупка жилья   недвижимость  
3  дополнительное образование    образование  
4             сыграть свадьбу        свадьба  
5               покупка жилья   недвижимость  
6           операции с жильем   недвижимость  
7                 образование    образование  
8       на проведение свадьбы        свадьба  
9     покупка жилья для семьи   недвижимость

**Вывод**

* Выделены леммы в значениях столбца с целями получения кредита, получилось 4 категории:
1. недвижимость
2. автомобиль
3. образование
4. свадьба


### Категоризация данных

Для отвена на вопрос: Есть ли зависимость между уровнем дохода и возвратом кредита в срок? нужно категоризировать уровень дохода.

Поделим всех клиентов на 4 категории с равным количеством человек с помощью 3 уровней дохода.

In [27]:
lvl_2 = clients['total_income'].median() # уровень 2 делит всех клиентов на две равные части
lvl_1 = clients[clients['total_income'] < lvl_2]['total_income'].median() # уровень 1 делит всех клиентов чей доход меньше уровня 2 на две равные части
lvl_3 = clients[clients['total_income'] > lvl_2]['total_income'].median() # уровень 3 делит всех клиентов чей доход больше уровня 2 на две равные части

In [28]:
def cat_group(income): # создаём функцию  
    if income < lvl_1:
        result = 'малоимущие'
    elif  lvl_1 <= income < lvl_2:
        result = 'средний класс'
    elif  lvl_2 <= income < lvl_3:
        result = 'состоятельные'
    else:
        result = 'богатые'
    return result

In [29]:
clients['class_income'] = clients['total_income'].apply(cat_group) # создаём столбец class_income и присваеваем категории 

/tmp/ipykernel_173/3535519125.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  clients['class_income'] = clients['total_income'].apply(cat_group) # создаём столбец class_income и присваеваем категории


In [30]:
def cat_children(child): # создаём функцию  
    if child == 0:
        result = 'нет детей'
    else:
        result = 'есть дети'
    return result

In [31]:
clients['cat_children'] = clients['children'].apply(cat_children) # создаём столбец cat_children и присваеваем категории 

In [32]:
display(clients.head(10)) # получение первых 10 строк таблицы clients

children  days_employed  dob_years education  education_id  \
0         1           8437         42    высшее             0   
1         1           4024         36   среднее             1   
2         0           5623         33   среднее             1   
3         3           4124         32   среднее             1   
4         0          14177         53   среднее             1   
5         0            926         27    высшее             0   
6         0           2879         43    высшее             0   
7         0            152         50   среднее             1   
8         2           6929         35    высшее             0   
9         0           2188         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0           253   
1   женат / замужем                 0      F   сотрудник     0           112   
2   женат / замужем                 0      M   сотрудник     0           145   
3   женат / замужем                 0      M   сотрудник     0           267   
4  гражданский брак                 1      F   пенсионер     0           158   
5  гражданский брак                 1      M   компаньон     0           255   
6   женат / замужем                 0      F   компаньон     0           240   
7   женат / замужем                 0      M   сотрудник     0           135   
8  гражданский брак                 1      F   сотрудник     0            95   
9   женат / замужем                 0      M   сотрудник     0           144   

                      purpose purpose_lemmas   class_income cat_children  
0               покупка жилья   недвижимость        богатые    есть дети  
1     приобретение автомобиля     автомобиль  средний класс    есть дети  
2               покупка жилья   недвижимость  состоятельные    нет детей  
3  дополнительное образование    образование        богатые    есть дети  
4             сыграть свадьбу        свадьба  состоятельные    нет детей  
5               покупка жилья   недвижимость        богатые    нет детей  
6           операции с жильем   недвижимость        богатые    нет детей  
7                 образование    образование  средний класс    нет детей  
8       на проведение свадьбы        свадьба     малоимущие    есть дети  
9     покупка жилья для семьи   недвижимость  состоятельные    нет детей

**Вывод**

* Созданы 4 категорий по уравню дохода:
1. малоимущие
2. средний класс
3. состоятельные
4. богатые

## Шаг 3. Ответы на вопросы

Для ответов на вопросы определим функцию, создающую сводную таблицу.

In [33]:
def pivot(col): # создаём функцию 
    clients_pivot = clients.pivot_table(index =col, columns ='debt', values = 'total_income',aggfunc='count') # создаём сводную таблицу
    clients_pivot.columns = ['no_debt', 'debt'] # присваиваем колонкам новые названия 
    clients_pivot['%'] =  clients_pivot['debt']/(clients_pivot['debt']+clients_pivot['no_debt'])*100 # создаём столбец с процентным соотношением невозвратных кредитов
    display(clients_pivot.sort_values(by = '%')) # выводим таблицу на экран

- Есть ли зависимость между наличием детей и возвратом кредита в срок?

In [34]:
pivot('cat_children') # вызываем функцию pivot с аргументом 'cat_children'

no_debt  debt         %
cat_children                         
нет детей       13028  1063  7.543822
есть дети        6685   678  9.208203

**Вывод**

Наличие детей уменьшает вероятность возврата кредита в срок.

- Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [35]:
pivot('family_status') # вызываем функцию pivot с аргументом 'family_status'

no_debt  debt         %
family_status                                 
вдовец / вдова             896    63  6.569343
в разводе                 1110    85  7.112971
женат / замужем          11408   931  7.545182
гражданский брак          3763   388  9.347145
Не женат / не замужем     2536   274  9.750890

**Вывод**

Семейное положение влияет на вероятность возврата кредита в срок, люди никогда не находившиеся в браке по кредиту платят хуже.

- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [36]:
pivot('class_income') # вызываем функцию pivot с аргументом 'class_income'

no_debt  debt         %
class_income                          
богатые           4971   383  7.153530
малоимущие        4451   383  7.923045
средний класс     4446   419  8.612539
состоятельные     5845   556  8.686143

**Вывод**

Зависимости от уровня дохода и возвратом кредита в срок нет. Можно выделить только богатых они по кредитам платят лучше.

- Как разные цели кредита влияют на его возврат в срок?

In [37]:
pivot('purpose_lemmas') # вызываем функцию pivot с аргументом 'purpose_lemmas'

no_debt  debt         %
purpose_lemmas                         
недвижимость      10029   782  7.233373
свадьба            2138   186  8.003442
образование        3643   370  9.220035
автомобиль         3903   403  9.359034

**Вывод**

Кредит на жильё и недвижимость чаще выплачивают в срок, чем на образование и автомобили.

## Шаг 4. Общий вывод

Лучше всего по кредитам платят богатые вдовы без детей покупающие жильё и недвижимость.
Хуже всего по кредитам платят холостые с детьми средний класс покупающие автомобиль.